robottelo/helpers.py

```diff
@@ -413,3 +414,86 @@ def default_url_on_new_port(oldport, newport):
 def get_func_name(func):
     """Given a func object return standardized name to use across project"""
     return '{0}.{1}'.format(func.__module__, func.__name__)
+
+
+def form_repo_path(org=None, lce=None, cv=None, cvv=None, prod=None,
+                   repo=None):
+    """Forms unix path to the directory containing published repository in
+    pulp using provided entity names. Supports both repositories in content
+    view version and repositories in lifecycle environment. Note that either
+    `cvv` or `lce` is required.
+
+    :param str org: organization label
+    :param str optional lce: lifecycle environment label
+    :param str cv: content view label
+    :param str optional cvv: content view version, e.g. '1.0'
+    :param str prod: product label
+    :param str repo: repository label
+    :return: full unix path to the specific repository
+    :rtype: str
+    """
+    if not all([org, cv, prod, repo]):
+        raise ValueError(
+            '`org`, `cv`, `prod` and `repo` arguments are required')
+    if not any([lce, cvv]):
+        raise ValueError('Either `lce` or `cvv` is required')
+
+    base_path = '/var/lib/pulp/published/yum/http/repos'
+    if lce:
+        repo_path = '{}/{}/{}/custom/{}/{}'.format(org, lce, cv, prod, repo)
+    elif cvv:
+        repo_path = '{}/content_views/{}/{}/custom/{}/{}'.format(
+            org, cv, cvv, prod, repo)
+
+    return '{}/{}'.format(base_path, repo_path)
+
+
+def create_repo(repo_fetch_url, packages, repo_name, hostname=None):
+    """Creates a repository from given packages and publishes it into pulp's
+    directory for web access.
+
+    :param str repo_fetch_url: URL to fetch packages from
+    :param packages: list of packages to fetch (with extension)
+    :param str repo_name: repository name - name of a directory with packages
+        and repodata
+    :param str optional hostname: hostname or IP address of the remote host. If
+        ``None`` the hostname will be get from ``main.server.hostname`` config.
+    :return: URL where the repository can be accessed
+    :rtype: str
+    """
+    repo_path = '/var/lib/pulp/published/yum/http/repos'
+    result = ssh.command(
+        'mkdir -p {}/{}'.format(repo_path, repo_name), hostname=hostname)
```

Comment:
change naming:
```python
# this can be moved to constants as PULP_PUBLISHED_REPOS_PATH
repos_path = '/var/lib/pulp/published/yum/http/repos'
repo_path = os.path.join(repos_path, repo_name)
result = ssh.command(
        'mkdir -p {}}'.format(repo_path), hostname=hostname)

```